File: udb_connector.php
<?php
class UDB_connector {
	// @var $cachedb -> Lugar donde se guarda la consulta (por defecto = cache.ucb)
	var $cachedb ="cache.ucb";
	// @var $udb_ac ARRAY -> Conexiones activas: $udb_ac['udbd.udd'] = BOOL
	var $udb_ac = array();
	// rac: read and cache
	// @param pdb File to rac
	function rac ($pdb) {
		//Open file for reading
		$rpdb = fopen($pdb, 'rb');
		//Open cache database 
		$rcachedb = fopen($this->cachedb, 'w');
		while (!feof($rpdb)) {
			fwrite($rcachedb, fgets($rpdb));
		}
		fclose($rcachedb);
		fclose($rpdb);
		return true;
	}
	// wac: write and cache
	// @param: pdb File to wac
	function wac ($pdb) {
		//Open file for writing
		$rpdb = fopen($pdb, 'wb');
		//Open cache database
		$rcachedb = fopen($this->cachedb, 'rb');
		while (!feof($rcachedb)) {
			fwrite($rpdb, fgets($rcachedb));
		}
		fclose($rcachedb);
		fclose($rpdb);
		return true;
	}
	
	// save: alias for wac
	// @param: pdb File to wac
	function save ($pdb) {
		$this->wac($pdb);
	}
	
	// connect: Conecta a una UDB
	// @param fdb File DataBase
	// @param nocache bool No guardar en la caché
	function connect ($fdb, $nocache = false) {
		if ($nocache == true) {
			if (file_exists($fdb)) {
				$this->udb_ac[$fdb] = true;
				return true;
			} else {
				$this->udb_ac[$fdb] = false;
				return false;
			}
		} else {
			if (!file_exists($fdb)) {
				$this->udb_ac[$fdb] = false;
				return false;
			} else {
				$this->rac($fdb);
				$this->udb_ac[$fdb] = true;
				return true;
			}
		}
	}
	
	// function: desconecta de una UDB
}

File: shiva/udb_connector.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
import os.path

class UNDB_Connector:
   def __init__(self):
      self.cachedb = "cache.ucb"
   # rac: read-and-cache
   def rac(self, pdb):
      rpdb = open(pdb, 'rb')
      rcachedb = open(self.cachedb, 'wb')
      for line in rpdb:
         rcachedb.write(line)
      rpdb.close()
      rcachedb.close()
      return True
   
   # wac: write-and-cache
   def wac(self, pdb):
      rpdb = open(pdb, 'wb')
      rcachedb = open(self.cachedb, 'rb')
      for line in rcachedb:
         rpdb.write(line)
      rpdb.close()
      rcachedb.close()
      return True

   # save: alias for wac
   def save(self, pdb):
      self.wac(pdb)
      
   # connect: connect to an undb
   def connect(self, fdb, nocache = False):
      if os.path.isfile(fdb):
         self.rac(fdb)
         return True
      else:
         return False
